oci/spec_opts.go

```diff
@@ -33,3 +34,51 @@ func WithHostname(name string) SpecOpts {
 		return nil
 	}
 }
+
+// WithEnv appends environment variables
+func WithEnv(environmnetVariables []string) SpecOpts {
+	return func(_ context.Context, _ Client, _ *containers.Container, s *specs.Spec) error {
+		if len(environmnetVariables) > 0 {
+			s.Process.Env = replaceOrAppendEnvValues(s.Process.Env, environmnetVariables)
+		}
+		return nil
+	}
+}
+
+// replaceOrAppendEnvValues returns the defaults with the overrides either
+// replaced by env key or appended to the list
+func replaceOrAppendEnvValues(defaults, overrides []string) []string {
```

Comment:
I think we should have unit tests now

Reply:
I added tests, but it is failing on Windows. Not sure why it isn't working, since I didn't write that bit.

I will boot into my Windows machine to see what is going on.

Reply:
Looks like the test was failing because the PATH wasn't being set by default on Windows, like it was on Linux. Maybe that is another issue that needs to be addressed?